### PR TITLE
Feature/preview

### PR DESCRIPTION
**Pull request by Bakunova-Olya**
add blocks personal information, languages, skills, certifications

### CODE REVIEW

File: src/containers/main-page/cv-form/components/fields/projects/components/SkillsToolsDialog.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import {
-  useState, memo, useEffect,
+  useState, memo,
 } from 'react';
```
Comment (alexMbicycle) [inline src/containers/main-page/cv-form/components/fields/projects/components/SkillsToolsDialog.tsx]: to single line please

File: src/containers/main-page/preview/Certifications.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+import ArticleIcon from '@mui/icons-material/Article';
+import { useUserFromDb } from '../cv-form/components/fields/personal-information/lib/query-hooks';
+
+const Certifications = function (): JSX.Element {
+  const { data, refetch } = useUserFromDb();
+  const isFetching = useIsFetching('db-user');
+  useEffect(() => {
+    if (isFetching) { refetch(); }
+  }, [refetch, isFetching]);
+  return (
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/Certifications.tsx]: Ask @varaMbicycle hoe to divide logically blocks of code. Spacing reqired.

File: src/containers/main-page/preview/Certifications.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    <Paper
+      elevation={2}
+      sx={{
+        maxWidth: '100%',
+        padding: '1.6rem',
+        margin: '1.8rem 2.4rem 0',
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/Certifications.tsx]: theme.spacing(...)

File: src/containers/main-page/preview/Certifications.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        borderRadius: '0.2rem',
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/Certifications.tsx]: theme.shape.borderRadius

File: src/containers/main-page/preview/Certifications.tsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        container
+        spacing={1}
+      >
+        <Box
+          sx={{
+            width: '3rem',
+            height: '3rem',
+            backgroundColor: 'secondary.main',
+            borderRadius: '50%',
+
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/Certifications.tsx]: no empty lines

File: src/containers/main-page/preview/Certifications.tsx [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+      <Grid
+        container
+        spacing={1}
+      >
+        <Box
+          sx={{
+            width: '3rem',
+            height: '3rem',
+            backgroundColor: 'secondary.main',
+            borderRadius: '50%',
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/Certifications.tsx]: replace to styled and set width and height in px

File: src/containers/main-page/preview/Certifications.tsx [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+
+          }}
+        >
+          <ArticleIcon color="primary" sx={{ margin: '0.7rem' }} />
+        </Box>
+        <Grid item xs={11}>
+          <Typography
+            variant="h5"
+            sx={{
+              marginLeft: '0.8rem',
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/Certifications.tsx]: call function inside and set theme.spacing or better replace to styled

File: src/containers/main-page/preview/Certifications.tsx [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+              fontWeight: 'fontWeightBold',
+            }}
+          >
+            CERTIFICATIONS
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/Certifications.tsx]: All literals store in constants.ts

File: src/containers/main-page/preview/Certifications.tsx [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+          </Typography>
+        </Grid>
+        <Grid item xs={8} />
+        <Grid
+          item
+          xs={4}
+          sx={{ textAlign: 'center' }}
+        >
+          <Typography variant="h5" color="text.disabled" align="center">
+            Date
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/Certifications.tsx]: as previous

File: src/containers/main-page/preview/Certifications.tsx [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+              <Typography
+                key={`${certificate.id}`}
+                sx={{ paddingLeft: '0.6rem' }}
+              >
+                <CircleIcon
+                  color="primary"
+                  sx={{
+                    width: '0.6rem',
+                    paddingTop: '0.6rem',
+                    marginRight: '0.6rem',
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/Certifications.tsx]: repalce to styled

File: src/containers/main-page/preview/Certifications.tsx [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+                  }}
+                />
+                <Link href={certificate.link}>
+                  {certificate.name}
+                </Link>
+
+              </Typography>
+            </Grid>
+            <Grid item xs={4} sx={{ textAlign: 'center' }}><Typography>{certificate.id}</Typography></Grid>
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/Certifications.tsx]: Look how plased Typography component, Failing code readability

File: src/containers/main-page/preview/Languages.tsx [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+import { useUserFromDb } from '../cv-form/components/fields/personal-information/lib/query-hooks';
+import RatingLanguage from './RatingLanguage';
+
+const Languages = function (): JSX.Element {
+  const { data, refetch } = useUserFromDb();
+  const isFetching = useIsFetching('db-user');
+  useEffect(() => {
+    if (isFetching) { refetch(); }
+  }, [refetch, isFetching]);
+  return (
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/Languages.tsx]: Spaces

File: src/containers/main-page/preview/Languages.tsx [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+    <Paper
+      elevation={2}
+      sx={{
+        maxWidth: '100%',
+        padding: '1.6rem',
+        margin: '18px 24px 0',
+        borderRadius: '0.2rem',
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/Languages.tsx]: to styled

File: src/containers/main-page/preview/Languages.tsx [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+      }}
+    >
+      <Grid
+        container
+        spacing={1}
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/Languages.tsx]: if there are two props please place them in one line
`<Grid container spacing={1} />`

File: src/containers/main-page/preview/Languages.tsx [hunk 15]
```diff
@@ -0,0 +1,88 @@
...
+      >
+        <Box
+          sx={{
+            width: '3rem',
+            height: '3rem',
+            backgroundColor: 'secondary.main',
+            borderRadius: '50%',
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/Languages.tsx]: to styled

File: src/containers/main-page/preview/Languages.tsx [hunk 16]
```diff
@@ -0,0 +1,88 @@
...
+
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/Languages.tsx]: no empty line

File: src/containers/main-page/preview/Languages.tsx [hunk 17]
```diff
@@ -0,0 +1,88 @@
...
+        >
+          <MenuBookIcon color="primary" sx={{ margin: '0.7rem' }} />
+        </Box>
+        <Grid item xs={11}>
+          <Typography
+            variant="h5"
+            sx={{
+              marginLeft: '0.8rem',
+              fontWeight: 'fontWeightBold',
+            }}
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/Languages.tsx]: to styled

File: src/containers/main-page/preview/Languages.tsx [hunk 18]
```diff
@@ -0,0 +1,88 @@
...
+          >
+            LANGUAGES
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/Languages.tsx]: to constants

File: src/containers/main-page/preview/Languages.tsx [hunk 19]
```diff
@@ -0,0 +1,88 @@
...
+          </Typography>
+        </Grid>
+        <Grid item xs={8} />
+        <Grid
+          item
+          xs={4}
+          sx={{ textAlign: 'center' }}
+        >
+          <Typography variant="h5" color="text.disabled">
+            Level
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/Languages.tsx]: to constants

File: src/containers/main-page/preview/Languages.tsx [hunk 20]
```diff
@@ -0,0 +1,88 @@
...
+          <Grid container sx={{ padding: '0 8px' }}>
+            <Grid item xs={2}>
+              <Typography key={language.name} sx={{ paddingLeft: '6px' }}>
+                <CircleIcon
+                  color="primary"
+                  sx={{
+                    width: '6px',
+                    paddingTop: '6px',
+                    marginRight: '6px',
+                  }}
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/Languages.tsx]: to styled

File: src/containers/main-page/preview/Languages.tsx [hunk 21]
```diff
@@ -0,0 +1,88 @@
...
+                />
+                {language.name}
+              </Typography>
+            </Grid>
+            <Grid item xs={6} sx={{ margin: 'auto' }}>
+              <RatingLanguage level={language.level} />
+            </Grid>
+            <Grid item xs={4} sx={{ textAlign: 'center' }}><Typography>{language.level}</Typography></Grid>
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/Languages.tsx]: Incorrect elements position

File: src/containers/main-page/preview/PersonalDetails.tsx [hunk 22]
```diff
@@ -0,0 +1,33 @@
...
+
+const PersonalDetails = function (): JSX.Element {
+  return (
+    <Grid
+      container
+      sx={{
+        maxWidth: '100%',
+        margin: 'auto',
+      }}
+    >
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/PersonalDetails.tsx]: You can use 
`<Container maxWidth="lg" /> `
this Grid instead
Also as I remenber you can add `component={Grid}` prop

File: src/containers/main-page/preview/PersonalInformation.tsx [hunk 23]
```diff
@@ -0,0 +1,89 @@
...
+
+const PersonalInformation = function (): JSX.Element {
+  const { data, refetch } = useUserFromDb();
+  const { photo } = useUserPhoto();
+  const isFetching = useIsFetching('db-user');
+  useEffect(() => {
+    if (isFetching) { refetch(); }
+  }, [refetch, isFetching]);
+  return (
+    <Paper
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/PersonalInformation.tsx]: Spacings

File: src/containers/main-page/preview/PersonalInformation.tsx [hunk 24]
```diff
@@ -0,0 +1,89 @@
...
+      elevation={2}
+      sx={{
+        maxWidth: '100%',
+        margin: '0 2.4rem',
+        padding: '1.6rem',
+        borderRadius: '0.2rem',
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/PersonalInformation.tsx]: to styled

File: src/containers/main-page/preview/PersonalInformation.tsx [hunk 25]
```diff
@@ -0,0 +1,89 @@
...
+      }}
+    >
+      <Grid
+        container
+        spacing={2}
+      >
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/PersonalInformation.tsx]: single line

File: src/containers/main-page/preview/PersonalInformation.tsx [hunk 26]
```diff
@@ -0,0 +1,89 @@
...
+        <Grid item xs={2}>
+          <ImageStyled
+            referrerPolicy="no-referrer"
+            alt="User"
+            src={photo}
+            sx={{
+              width: '6.8rem',
+              height: '7.2rem',
+            }}
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/PersonalInformation.tsx]: styled

File: src/containers/main-page/preview/PersonalInformation.tsx [hunk 27]
```diff
@@ -0,0 +1,89 @@
...
+          <Typography
+            variant="caption"
+            align="center"
+          >
+            Work Experience
+          </Typography>
+        </Grid>
+        <Grid item xs={12}>
+          <Typography variant="h6" fontWeight="fontWeightBold">
+            SUMMARY OF QUALIFICATION
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/PersonalInformation.tsx]: to constants

File: src/containers/main-page/preview/RatingLanguage.tsx [hunk 28]
```diff
@@ -0,0 +1,56 @@
...
+  'Pre-Intermediate': 2,
+  Intermediate: 3,
+  'Upper-Intermediate': 4,
+  Advanced: 5,
+  Proficiency: 6,
+};
+
+const RatingLanguage = function (props: { [index : string] : string }): JSX.Element {
+  const { level } = props;
+  return (
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/RatingLanguage.tsx]: spacing required

File: src/containers/main-page/preview/RatingLanguage.tsx [hunk 29]
```diff
@@ -0,0 +1,56 @@
...
+    <Box
+      sx={{
+
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/RatingLanguage.tsx:7]: no empty line

File: src/containers/main-page/preview/RatingLanguage.tsx [hunk 30]
```diff
@@ -0,0 +1,56 @@
...
+        display: 'flex',
+        justifyContent: 'center',
+        alignItems: 'center',
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/RatingLanguage.tsx]: to styled

File: src/containers/main-page/preview/RatingLanguage.tsx [hunk 31]
```diff
@@ -0,0 +1,56 @@
...
+        precision={1}
+        readOnly
+        icon={(
+          <RectangleBlueIcon
+            fontSize="large"
+            color="primary"
+            sx={{
+              paddingRight: '0.2rem',
+              width: '3.6rem',
+            }}
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/RatingLanguage.tsx]: To styled

File: src/containers/main-page/preview/RatingLanguage.tsx [hunk 32]
```diff
@@ -0,0 +1,56 @@
...
+          />
+        )}
+        emptyIcon={(
+          <RectangleIcon
+            fontSize="large"
+            sx={{
+              paddingRight: '0.2rem',
+              width: '3.6rem',
+            }}
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/RatingLanguage.tsx]: same as previous

File: src/containers/main-page/preview/RatingLanguage.tsx [hunk 33]
```diff
@@ -0,0 +1,56 @@
...
+          />
+        )}
+      />
+    </Box>
+  );
+};
+export default RatingLanguage;
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/RatingLanguage.tsx]: You are using props, memoizing required.

File: src/containers/main-page/preview/Skills.tsx [hunk 34]
```diff
@@ -0,0 +1,115 @@
...
+  }, [refetch, isFetching]);
+  return (
+    <Paper
+      elevation={1}
+      sx={{
+        maxWidth: '100%',
+        padding: '1.6rem',
+        margin: '1.8rem 2.4rem 0',
+        borderRadius: '0.2rem',
+      }}
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/Skills.tsx]: to styled

File: src/containers/main-page/preview/Skills.tsx [hunk 35]
```diff
@@ -0,0 +1,115 @@
...
+    >
+      <Grid
+        container
+        spacing={1}
+      >
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/Skills.tsx]: Single line

File: src/containers/main-page/preview/Skills.tsx [hunk 36]
```diff
@@ -0,0 +1,115 @@
...
+        <Box
+          sx={{
+            width: '3rem',
+            height: '3rem',
+            backgroundColor: 'secondary.main',
+            borderRadius: '50%',
+
+          }}
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/Skills.tsx]: to styled

File: src/containers/main-page/preview/Skills.tsx [hunk 37]
```diff
@@ -0,0 +1,115 @@
...
+        >
+          <SchoolIcon color="primary" sx={{ margin: '0.7rem' }} />
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/Skills.tsx]: to styled

File: src/containers/main-page/preview/Skills.tsx [hunk 38]
```diff
@@ -0,0 +1,115 @@
...
+        </Box>
+        <Grid item xs={11}>
+          <Typography
+            variant="h5"
+            sx={{
+              marginLeft: '0.8rem',
+              fontWeight: 'fontWeightBold',
+            }}
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/Skills.tsx]: to styled

File: src/containers/main-page/preview/Skills.tsx [hunk 39]
```diff
@@ -0,0 +1,115 @@
...
+          >
+            SKILLS
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/Skills.tsx]: to constants

File: src/containers/main-page/preview/Skills.tsx [hunk 40]
```diff
@@ -0,0 +1,115 @@
...
+                  <Grid container sx={{ padding: '0 0.8rem' }}>
+                    <Grid item xs={2}>
+                      <Typography key={tool.name}>
+                        <CircleIcon
+                          color="primary"
+                          sx={{
+                            width: '0.6rem',
+                            paddingTop: '0.6rem',
+                            marginRight: '0.6rem',
+                          }}
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/Skills.tsx]: to styled

File: src/containers/main-page/preview/TopBox.tsx [hunk 41]
```diff
@@ -0,0 +1,79 @@
...
+
+const TopBox = function (): JSX.Element {
+  const { data, refetch } = useUserFromDb();
+  const isFetching = useIsFetching('db-user');
+  useEffect(() => {
+    if (isFetching) { refetch(); }
+  }, [refetch, isFetching]);
+
+  return (
+
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/TopBox.tsx]: no empty row

File: src/containers/main-page/preview/TopBox.tsx [hunk 42]
```diff
@@ -0,0 +1,79 @@
...
+    <Grid
+      container
+      spacing={0}
+      sx={{
+        height: '16.8rem',
+        backgroundColor: 'primary.main',
+      }}
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/TopBox.tsx]: to styled

File: src/containers/main-page/preview/TopBox.tsx [hunk 43]
```diff
@@ -0,0 +1,79 @@
...
+  const isFetching = useIsFetching('db-user');
+  useEffect(() => {
+    if (isFetching) { refetch(); }
+  }, [refetch, isFetching]);
+
+  return (
+
+    <Grid
+      container
+      spacing={0}
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/TopBox.tsx]: remove. Zero by default

File: src/containers/main-page/preview/TopBox.tsx [hunk 44]
```diff
@@ -0,0 +1,79 @@
...
+      sx={{
+        height: '16.8rem',
+        backgroundColor: 'primary.main',
+      }}
+    >
+      <Grid
+        item
+        xs={4}
+        sx={{
+
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/TopBox.tsx]: no empty row and replace to styled.ts

File: src/containers/main-page/preview/TopBox.tsx [hunk 45]
```diff
@@ -0,0 +1,79 @@
...
+        }}
+      >
+
+        <LogoIcon style={{
+          width: '12rem',
+          height: '2.8rem',
+          margin: '1rem',
+          fill: 'secondary',
+        }}
+        />
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/TopBox.tsx]: Styled

File: src/containers/main-page/preview/index.tsx [hunk 46]
```diff
@@ -0,0 +1,19 @@
+import { useIsFetching } from 'react-query';
+import { useEffect } from 'react';
+import { useUserFromDb } from '../cv-form/components/fields/personal-information/lib/query-hooks';
+import Wrapper from './Wrapper';
+
+const Preview = function (): JSX.Element {
+  const { refetch } = useUserFromDb();
+  const isFetching = useIsFetching('db-user');
+  useEffect(() => {
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/index.tsx]: Spacings

File: src/containers/main-page/preview/index.tsx [hunk 47]
```diff
@@ -0,0 +1,19 @@
...
+    if (isFetching) { refetch(); }
+  }, [refetch, isFetching]);
+
+  return (
+    <Wrapper />
+
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/index.tsx]: no empty line

**Review by alexMbicycle — CHANGES_REQUESTED** [review]
Remove all sx to styled just if is in the sx a single parameter you can leave it.

**Comment by Bakunova-Olya** [comment]
All bug fixed

File: src/containers/main-page/preview/Certifications.tsx [hunk 48]
```diff
@@ -0,0 +1,71 @@
...
+    if (isFetching) refetch();
+  }, [refetch, isFetching]);
+
+  return (
+    <PaperWrapperStyled
+      elevation={1}
+    >
+      <Grid container>
+        <BoxWrapperStyled>
+          <ArticleIcon color="primary" sx={{ margin: theme.spacing(1.75) }} />
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/Certifications.tsx]: image

Please change everywhere. Using extra dependencies is not good approach

File: src/containers/main-page/preview/RatingLanguage.tsx [hunk 49]
```diff
@@ -0,0 +1,32 @@
+import {
+  Rating,
+} from '@mui/material';
+import React from 'react';
+import { Labels } from '../cv-form/components/fields/languages/components/utils/level.enum';
+import { BoxRatingStyled, RectangleBlueIconStyled, RectangleIconStyled } from '../styled';
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/RatingLanguage.tsx]:
```
import React from 'react';
import { Rating } from '@mui/material';

import { Labels } from '../cv-form/components/fields/languages/components/utils/level.enum';

import { BoxRatingStyled, RectangleBlueIconStyled, RectangleIconStyled } from '../styled';
```

File: src/containers/main-page/preview/Skills.tsx [hunk 50]
```diff
@@ -0,0 +1,90 @@
+import { useIsFetching } from 'react-query';
+import { useEffect } from 'react';
+
+import {
+  Grid, Typography,
+} from '@mui/material';
+
+import { useUserFromDb } from '../cv-form/components/fields/personal-information/lib/query-hooks';
+import theme from '../../../common/theme';
+import {
```
Comment (alexMbicycle) [inline src/containers/main-page/preview/Skills.tsx]:
```
import { useEffect } from 'react';
import { useIsFetching } from 'react-query';

import { Grid, Typography } from '@mui/material';

import theme from '../../../common/theme';
import { useUserFromDb } from '../cv-form/components/fields/personal-information/lib/query-hooks';
import { CV_FORM_STEPS } from '../cv-form/utils/constants';
import {
  PaperWrapperStyled, BoxWrapperStyled, CircleIconStyled,
   SectionTitle, SkillsGrid, SchoolIconStyled,
} from '../styled';

```

**Review by alexMbicycle — CHANGES_REQUESTED** [review]

**Comment by Bakunova-Olya** [comment]
All bug fixed

**Review by alexMbicycle — APPROVED** [review]

**Pull request merged**